### PR TITLE
corrected logic for code challenge so two linked lists merge all thei…

### PR DESCRIPTION
**Pull request by SharinaS**
…r nodes into one list. Wrote more tests.

- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [ ] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

### CODE REVIEW

File: code401Challenges/src/main/java/code401Challenges/linkedlist/LinkedList.java [hunk 1]
```diff
@@ -172,5 +172,31 @@ public static LinkedList mergeLists(LinkedList a, LinkedList b) {
         return b;
     }
 
+    // Merge two linked lists
+    // Received help with logic from Michelle Ferreirae
+    public static LinkedList mergeLinkedLists(LinkedList a, LinkedList b) {
+        if (a.head == null) {
+            return b;
+        }
```
Comment (jeffborda) [inline code401Challenges/src/main/java/code401Challenges/linkedlist/LinkedList.java]: This is a good check (if list a is empty, then just return the other list). However, if list b is empty the algorithm will break. So you could add another check for if list b is empty, return a.
